ra2ce/analyses/indirect/losses.py

```diff
@@ -125,91 +161,145 @@ def calc_vlh(
+                partofday_trip_purpose_name = self.partofday + f"_{purpose}"
+                partofday_trip_purpose_intensity_name = "intensity_" + partofday_trip_purpose_name
+                # read and set the vot's
+                globals()[vot_var_name] = \
+                    self.values_of_time.loc[self.values_of_time["trip_types"] == purpose, "value_of_time"].item()
+                # read and set the intensities
+                globals()[partofday_trip_purpose_intensity_name] = self.intensities[partofday_trip_purpose_name] / 24
+        # shape vlh
+        vlh = pd.DataFrame(
+            index=self.intensities.index,  # "link_type"
+        )
+        events = self.criticality_data.filter(regex='^EV')
+        # Read the performance_change stating the functionality drop
+        performance_change = self.criticality_data[self.performance_metric]
+
+        # find the link_type and the inundation height
+        vlh = pd.merge(vlh, self.network[['link_id', 'link_type']], left_index=True, right_on='link_id')
+        vlh = pd.merge(vlh, self.criticality_data[['link_id'] + list(events.columns)],
+                       left_index=True, right_on='link_id')
+
+        # set vot values for trip_types
+        _set_vot_intensity_per_trip_purpose(trip_types)
+
+        # for each link and for each event calculate vlh
+        for link_id, vlh_row in vlh.iterrows():
+            for event in events:
+                vlh_event_total = 0
+                row_inundation_range = _get_range(vlh_row[event])
+                link_type_inundation_range = f"{vlh_row['link_type']}_{row_inundation_range}"
+
+                # get stepwise recovery curve data
+                relevant_curve = self.resilience_curve[
+                    self.resilience_curve["link_type_inundation_height"] == link_type_inundation_range
+                    ]
+                duration_steps: list = relevant_curve["duration_steps"].item()
+                functionality_loss_ratios: list = relevant_curve["functionality_loss_ratio"].item()
+
+                # get vlh_trip_type_event
+                for trip_type in trip_types:
+                    intensity_trip_type = globals().get(f"intensity_{trip_type}")
+                    vlh_trip_type_event = sum(
+                        intensity_trip_type * duration * loss_ratio * performance_change
+                        for duration, loss_ratio in zip(duration_steps, functionality_loss_ratios)
+                    )
+
+                    vlh[f"vlh_{trip_type}_{event}"] = vlh_trip_type_event
+                    vlh_event_total += vlh_trip_type_event
+                vlh[f"vlh_{event}_total"] = vlh_event_total
+
+        return vlh
+
+    def calculate_losses_from_table(self) -> pd.DataFrame:
+        """This function opens an existing table with traffic data and value of time to calculate losses based on
+        detouring values. It also includes
+            a traffic jam estimation.
+            """
```

Comment:
Please check your code formatting.
```suggestion
        """
        This function opens an existing table with traffic data and value of time to calculate losses based on
        detouring values. It also includes a traffic jam estimation.
        """
```